### PR TITLE
Added Circle CI config file

### PR DESCRIPTION
**Pull request by tilleryd**

### CODE REVIEW

File: packages/react-scripts/template/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
+
+# Set defaults for all jobs
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    CC_TEST_REPORTER_ID: ${CC_TEST_REPORTER_ID}
```
Comment (tilleryd) [inline packages/react-scripts/template/.circleci/config.yml]: Code Climate test reporter ID should be retrieved from an env var set via the Circle CI dashboard.

File: packages/react-scripts/template/.circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  test:linters:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Run linters
+          command: |
+            npm run lint-js -- --max-warnings=0 && npm run lint-css
```
Comment (spmcbride1201) [inline packages/react-scripts/template/.circleci/config.yml]: We need to add these package scripts and the associated deps.

File: packages/react-scripts/template/.circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          name: Build the application
+          command: |
+            npm run build
+      - run:
+          name: Compress the build directory into a tarball
+          command: |
+            tar -czf build.tar.gz build
+     # Store Assets generated by npm build
+      - store_artifacts:
+          path: bundlesize
```
Comment (spmcbride1201) [inline packages/react-scripts/template/.circleci/config.yml]: We need to configure webpack to use webpack-bundle-analyzer

File: packages/react-scripts/template/.circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      - checkout
+      - attach_workspace:
+          at: ~/repo
+      # CircleCI 2.0 no longer allows interpolation of env variables. This is the workaround:
+      # https://discuss.circleci.com/t/using-environment-variables-in-config-yml-not-working/14237
+      - run:
+          command: mkdir results
+      - run:
+          name: Set JEST_JUNIT_OUTPUT
+          command: echo 'export JEST_JUNIT_OUTPUT="results/junit-$CIRCLE_NODE_INDEX.xml"' >> $BASH_ENV
```
Comment (spmcbride1201) [inline packages/react-scripts/template/.circleci/config.yml]: we need to add the junit reporter and set jest to use it

File: packages/react-scripts/template/.circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      - run:
+          name: Prepare Code Climate Test Reporter
+          command: |
+            ./cc-test-reporter before-build
+      - run:
+          name: Run tests
+          # split tests by timing and then run npm test
+          command: |
+            circleci tests glob "src/**/*.test.js" | circleci tests split --split-by=timings | \
```
Comment (spmcbride1201) [inline packages/react-scripts/template/.circleci/config.yml]: we need to add the junit reporter dep

File: packages/react-scripts/template/.circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+          at: ~/repo
+      - run:
+          name: Send Istanbul LCOV Results to Code Climate
+          command: |
+            ./cc-test-reporter sum-coverage coverage/codeclimate.*.json
+            ./cc-test-reporter upload-coverage
+      - run:
+          name: Merge junit results
+          command: |
+            mkdir junit && npm run junit-merge -- -d results -o junit/junit.xml
```
Comment (spmcbride1201) [inline packages/react-scripts/template/.circleci/config.yml]: we need to install `junit-merge`

**Pull request closed without merging**